Libs/MRML/Core/vtkMRMLMeasurement.cxx

```diff
@@ -408,26 +490,51 @@ void vtkMRMLMeasurement::SetValue(double value, vtkMRMLUnitNode* unitNode, int l
     this->ValueDefined = true;
     modified = true;
     }
-  if (this->Units != units)
+  ComputationResult computationResult = vtkMRMLMeasurement::OK;
+  if (this->LastComputationResult != computationResult)
     {
-    this->Units = units;
+    this->LastComputationResult = computationResult;
     modified = true;
     }
-  if (this->PrintFormat != printFormat)
+  if (modified)
     {
-    this->PrintFormat = printFormat;
-    modified = true;
+    this->Modified();
+    }
+}
+
+//---------------------------------------------------------------------------
+double vtkMRMLMeasurement::GetDisplayValue()
+{
+  return this->Value * this->DisplayCoefficient;
+};
+
+//---------------------------------------------------------------------------
+vtkMRMLUnitNode* vtkMRMLMeasurement::GetUnitNode(const char* quantityName)
+{
+  if (!quantityName || strlen(quantityName)==0)
+    {
+    return nullptr;
     }
-  if (lastComputationResult != ComputationResult::NoChange)
+  if (!this->InputMRMLNode || !this->InputMRMLNode->GetScene())
```

Comment:
I think it is a valid scenario not having an input node for constant measurements.

Reply:
We need a reference to the scene to get the unit node from the scene. I'll add documentation to the method that unit can only be retrieved if an input node is set (it should be always possible to find a node that can be associated with a measurement) and a warning message if the input node is not set.